src/sentry/tasks/integrations.py

```diff
@@ -21,7 +21,7 @@
 )
 # TODO(jess): Add more retry exclusions once ApiClients have better error handling
 @retry(exclude=(ExternalIssue.DoesNotExist, Integration.DoesNotExist))
-def post_comment(external_issue_id, data, **kwargs):
+def post_comment(external_issue_id, data, user, **kwargs):
```

Comment:
This should probably be `user_id` for the same reasons we're doing `external_issue_id`. This data is being serialized into and stored somewhere for the message queue to pick off. It's best to keep these values as close to the absolute basic data structures possible (`int`, `string`, `dict`, etc.).

Reply:
@mnoble when should I do that? Is that always when I have a task? Or are there other times? I don't really know what happens here. I kind of assumed it would just pass a pointer to the object in memory... but if it's serializing and such... then it works a bit differently than I expect. Maybe we could talk about that a bit offline?

Reply:
Yea, any time you're enqueueing a task.

Okay, so imagine you're enqueueing the task above:

```
post_comment.apply_async(external_issue_id=1, data={}, user_id=1)
```

So this is going to take your function call and serialize it into a format it can store and then deserialize later. It does this because the Worker that will handle this job is a separate process (probably on a separate server). It doesn't share the same memory as the application that enqueued the job. So we need a way to go from

```
function call --> store on server that is not the app server --> function call for worker on another server
```

Most systems serialize that message and store it to disk somewhere (database, files, whatever). So you end up with a list of serialized tasks that are waiting for a worker to handle. Let's just assume the serialization is done via JSON. When you use IDs, you'll end up with:

```
{"task": "do_thing", "foo": "bar"}
{"task": "do_other_thing", "baz": "beep"}
{"task": "post_comment", "external_issue_id": 1, "data": {}, "user_id": 1}
```

So now imagine instead of passing model IDs, you pass the model object itself. You'd end up with these tasks in the queue (scroll to the right for full effect!):

```
{"task": "do_thing", "foo": "bar"}
{"task": "do_other_thing", "baz": "beep"}
{"task": "post_comment", "external_issue_id": 1, "data": {}, "user": {"id" : 1,"session_nonce" : null,"is_superuser" : true,"is_managed" : false,"last_password_change" : "2018-07-16 10:26:49.292936-07","last_active" : "2018-09-07 11:01:41.370316-07","is_sentry_app" : null,"is_active" : true,"is_password_expired" : false,"last_login" : "2018-09-07 11:02:13.36073-07","password" : "hjkl435jklh435jk43lk5jnbvjnvghc3g21=","first_name" : "","date_joined" : "2018-07-16 10:26:49.242432-07","username" : "matte@sentry.io","email" : "matte@sentry.io","is_staff" : true, "flags" : 0}}
```

It's going to need to serialize the entire object AND deserialize it on the other side when a worker picks it up. [De]serialization usually isn't cheap and even if it was, a database lookup is probably faster regardless. We also don't need to store all of this data to disk. We don't need most of it.

So using basic data structures makes it easier and more efficient for Celery to store the data as well as pass it to a different process on a different server to handle.